superset-frontend/src/explore/components/controls/FilterBoxItemControl.jsx

```diff
@@ -202,6 +220,21 @@ export default class FilterBoxItemControl extends React.Component {
             />
           }
         />
+        <FormRow
+          label={t('Fixed Filter Options')}
+          tooltip={t(
+            'By default, filter box will only load top 1000 filter options once. ' +
+              'When disabled, filter box will load options from a remote source as the user types, ' +
+              'with 1000 options per load.',
```

Comment:
Should we make the async loading the default or allow Superset admins to change it?

Reply:
let me check with PM tomorrow.

Reply:
!danger here, it's very possible that you'd be running dozens of `SELECT DISTINCT dimension_column FROM VERY_LARGE_TABLE WHERE dimension_column LIKE '%{str}%'`.  Your poor Presto cluster might suffer.

Also the cache hit rate would be very low compared to the normal filter.

Reply:
So I'd argue for unchecked by default, and a good explanation of the tradeoff in a toolitp